bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/basic/MiIoDatabaseWatchService.java

```diff
@@ -109,14 +112,11 @@ private void populateDatabase() {
         for (URL db : urlEntries) {
             logger.trace("Adding devices for db file: {}", db);
             try {
-                @Nullable
                 JsonObject deviceMapping = Utils.convertFileToJSON(db);
-                if (deviceMapping != null) {
-                    Gson gson = new GsonBuilder().serializeNulls().create();
-                    MiIoBasicDevice devdb = gson.fromJson(deviceMapping, MiIoBasicDevice.class);
-                    for (String id : devdb.getDevice().getId()) {
-                        workingDatabaseList.put(id, db);
-                    }
+                Gson gson = new GsonBuilder().serializeNulls().create();
```

Comment:
This Gson instance should be reused, but I guess you can do that in another PR.

Reply:
Thanks, moved it so it is reused indeed.